### ISSUE TITLE
Connect 2 DLLs to one BLE IndoorBike 

### ISSUE THREAD

**Issue by ShuohengZhang** [adabru/BleWinrtDll#63]
#26 Hi @adabru, I have fixed the DllNotFoundException by downloading the dlls that are missing. As only the write functions in BleWinrtDll do not work in my project, I want to use my DLL as a substitute. As I tested them separately, they worked well. However, when I use BleWinrtDll to subscribe to the 2ad2 (and ignore the 2ad9), I still cannot subscribe to the write function through my DLL. 

![Captur1e](https://github.com/adabru/BleWinrtDll/assets/143949385/eb81a35e-f4d3-4d16-a450-429e7c35eb98)
![Capt2ure](https://github.com/adabru/BleWinrtDll/assets/143949385/4229d8f8-0373-415f-b7e6-00bee3f798f3)
![Capt3ure](https://github.com/adabru/BleWinrtDll/assets/143949385/b7d81a6a-f9d3-4ea8-b208-17d78530a63a)
As the screenshot shows, it goes from line 218 to 220 and does not go in. 

At this position, I would like to know if there are any locks in your functions that prevent another app or program from subscribing to the same service? Thank you.

Best regards,
Shuoheng

**Comment by adabru** [comment]
Are you using both dll's simultaneously? Do you get an exception? If not, maybe the service has zero characteristics?

I had problems on getting a BLEDevice or Service multiple times from Windows and had to cache them in BleWinrtDll.cpp. Maybe caching the device/services will help?

**Comment by ShuohengZhang** [comment]
Yes I'm using them simultaneously, and I got no exception. I think ur right, the second dll subscribed to the service but has no characteristics to find and subscribe ( like blocked by the other one). 

Maybe I can try to cache them, but as I have seldom experience with build and test dlls, could you please explain with more detail so that I can have a try.

Thank you !

**Comment by adabru** [comment]
In line 211, instead of calling `device.GetGattServicesAsync())`, can you store the service from step 3.1 in a variable and reuse that variable here?
